src/react/hooks/auth.ts

```diff
@@ -1,6 +1,35 @@
 import { USER } from "../../apollo/queries";
+import {
+  Maybe,
+  UserFragment,
+  UserDetailsQuery,
+  UserDetailsQueryVariables,
+} from "../../apollo/types";
 import { hookFactory } from "../helpers/hookFactory";
 import { hookStateFactory } from "../helpers/hookStateFactory";
 
+type AuthState = {
+  user: Maybe<UserFragment> | undefined;
+  authenticated: boolean;
+};
+
+/**
+ * React hook to get authorization methods
+ *
+ * @returns Saleor's authorization methods
+ */
 export const useAuth = hookFactory("auth");
-export const useAuthState = hookStateFactory(USER);
+
+/**
+ * React hook to get user's authentication data.
+ *
+ * @returns Object with user's data
+ */
+export const useAuthState = (): AuthState => {
+  const { data } = hookStateFactory<
+    UserDetailsQuery,
+    UserDetailsQueryVariables
+  >(USER);
+
+  return { user: data?.me, authenticated: data?.authenticated ? true : false };
```

Comment:
Seems like: return { ...data } is good enough

Reply:
Data might be undefined but I guess that we always want to return `{ user: User; authenticated: boolean}`.

Reply:
right, in this case `!!data?.authenticated` should be good enough

Reply:
changed